forms-flow-web/src/components/Modeler/SubFlowTable.js

```diff
@@ -44,19 +44,31 @@ const SubFlow = React.memo(() => {
           processType: "BPMN",
           limit: limit,
           searchKey: search,
-          sortBy: Object.keys(sort),
-          sortOrder: Object.values(sort),
+          sortBy: currentBPMNsort.sortBy,
+          sortOrder: currentBPMNsort.sortOrder,
         },
         () => {
           setIsLoading(false);
           setSearchSubflowLoading(false);
         }
       )
     );
-  }, [dispatch, activePage, limit, searchText, sort]);
+  }, [dispatch, activePage, limit, searchText, currentBPMNsort]);
 
-  // const handleSort = (key,value)=> setSort(prev => ({...prev,[key]:value}));
-  const handleSort = (key, value) => setSort({ [key]: value });
+
+  const handleSort = (key) => {
+    setCurrentBPMNsort((prevSort) => {
+      let newSortOrder = "asc";
+
+      if (prevSort.sortBy === key) {
+        newSortOrder = prevSort.sortOrder === "asc" ? "desc" : "asc";
```

Comment:
here some confusion : here the code accepting only one and not keeping previous keys, so if we are accepting one key just  toggle to asc or desc

`const handleSort = (key) => {
  setCurrentBPMNsort(({ sortBy, sortOrder }) => ({
    sortBy: key,
    sortOrder: sortBy === key && sortOrder === "asc" ? "desc" : "asc",
  }));
};`

Reply:
Yes. if prevSort sortBy is same as to current  will update the sortorder. @shuhaib-aot